webapp/src/components/Analysis/UtterancesTable.tsx

```diff
@@ -390,6 +403,68 @@ const UtterancesTable: React.FC<Props> = ({
   ];
 
   const searchString = constructSearchString(pipeline);
+
+  const updateProposedAction = (data: UtterancePatch[]) => {
+    const groupByPersistentId: Record<
+      string,
+      UtterancePatch["persistentId"][]
+    > = data.reduce(
+      (
+        accumulated: Record<string, UtterancePatch["persistentId"][]>,
+        { persistentId, dataAction }: UtterancePatch
+      ) => ({
+        ...accumulated,
+        [dataAction]: accumulated[dataAction]
+          ? accumulated[dataAction].concat(persistentId)
+          : [persistentId],
+      }),
+      {}
+    );
+    Object.entries(groupByPersistentId).forEach(
+      ([dataAction, persistentIds]) => {
+        const newValue = dataAction as DataAction;
+        updateDataAction({
+          persistentIds,
+          newValue,
+          ...getUtterancesQueryState,
+        });
+      }
+    );
+  };
+
+  const importProposedActions = (file: File) => {
+    const fileReader = new FileReader();
+    fileReader.onload = ({ target }) => {
+      if (target) {
+        const result = target.result as string;
+        const [header, ...rows] = result.split("\n");
+        if (rows.join()) {
+          const headers: string[] = header.trim().split(",");
+          if (
+            [config.columns.persistent_id, "proposed_action"].every((h) =>
+              headers.includes(h)
+            )
+          ) {
+            const utterancePatch: UtterancePatch[] = rows.map((row) => {
+              const [persistentId, dataAction] = row.trim().split(",");
+              return { persistentId, dataAction } as UtterancePatch;
+            });
+            updateProposedAction(utterancePatch);
+          } else {
+            showDialogMessage(
+              "The CSV file did not have the row_idx and proposed_action column headers to update the proposed action."
+            );
+            showDialog(true);
+          }
+        } else {
+          showDialogMessage("There are no records in the CSV file.");
+          showDialog(true);
```

Comment:
Now that I see where the `Dialog` box is used, it seems to only be showing error messages. Right? Have you considered using `raiseErrorToast` instead? It seems more aligned with the rest of the app.
```suggestion
          raiseErrorToast("There are no records in the CSV file");
```

Reply:
So, do you suggest we use the "raiseErrorToast" for both scenarios?

Reply:
That's what I am contemplating, yes. What do you think?

Reply:
Yes, you are right; showing an error message is an excellent idea using toast.